src/components/pages/imports.js

```diff
@@ -0,0 +1,112 @@
+import by from 'thenby';
+import React from 'react';
+import Page from '../page.js';
+import PropTypes from 'prop-types';
+import urls from '../../utils/urls.js';
+import { useTranslation } from 'react-i18next';
+import { Table, Collapse, Button } from 'antd';
+import { useDateFormat } from '../language-context.js';
+import { importBatchShape, importSourceShape } from '../../ui/default-prop-types.js';
+
+const Panel = Collapse.Panel;
+
+function Imports({ initialState }) {
+  const { t } = useTranslation('imports');
+  const { formatDate } = useDateFormat();
+
+  const { batches, importSources } = initialState;
+  let sourceMap = importSources.reduce((acc, source) => {
+    acc[source.hostName] = { name: source.name, batches: [] };
+    return acc;
+  }, {});
+
+  sourceMap = batches.reduce((acc, batch) => {
+    const hostName = batch.batchParams.hostName;
+    acc[hostName] = {
+      name: batch.batchParams.name,
+      batches: [...acc[hostName].batches || [], batch]
+    };
+    return acc;
+  }, sourceMap);
+
+  const sources = Object.entries(sourceMap).map(([key, value]) => ({
+    importSourceHost: key,
+    importSourceName: value.name,
+    batches: value.batches.sort(by(batch => batch.createdOn, 'desc'))
+  }));
+
+  const handleCreateImport = source => {
+    window.location = urls.getCreateImportUrl(source.importSourceHost);
+  };
+
+  const renderId = id => <a href={urls.getBatchUrl(id)}>{t('viewBatch')}</a>;
+
+  const renderStatus = (_, batch) => {
```

Comment:
Also small, but can we put the name of the first argument here as `_name`? I think this helps alot for future development, in case that argument is needed, to directly see what is expected there and not have to dig for it.

Reply:
Actually for this one I realized that the batch itself is the "_" :)). Thanks!

Reply:
Oh, ouch 😄